api/fluentdoperator/v1alpha1/fluentd_types.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"kubesphere.io/fluentbit-operator/api/fluentdoperator/v1alpha1/plugins/input"
+	"kubesphere.io/fluentbit-operator/pkg/utils"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// FluentdSpec defines the desired state of Fluentd
+type FluentdSpec struct {
+	// Fluentd global inputs.
+	GlobalInputs []input.Input `json:"globalInputs,omitempty"`
+	// By default will build the related service according to the globalinputs definition.
+	DisableService bool `json:"disableService,omitempty"`
+	// Numbers of the Fluentd instance
+	Replicas *int32 `json:"replicas,omitempty"`
+	// Fluentd image.
+	Image string `json:"image,omitempty"`
+	// Fluentd Watcher command line arguments.
+	Args []string `json:"args,omitempty"`
+	// MatchCfgSelector defines the selectors to select the fluentd config CRs.
+	MatchCfgSelector metav1.LabelSelector `json:"matchCfgSelector,omitempty"`
+	// Buffer definition
+	BufferVolume *BufferVolume `json:"buffer,omitempty"`
+	// Fluentd image pull policy.
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
+	// Fluentd image pull secret
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+	// Compute Resources required by container.
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+	// NodeSelector
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// Pod's scheduling constraints.
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+	// Tolerations
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// RuntimeClassName represents the container runtime configuration.
+	RuntimeClassName string `json:"runtimeClassName,omitempty"`
```

Comment:
Do `fluentd` images have different configurations when different container runtime?

Reply:
Added to TODO list.

Reply:
no, it's a log aggregator, not a log agent. So it only receives logs via network.